pkg/crc/network/proxy.go

```diff
@@ -0,0 +1,201 @@
+package network
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/asaskevich/govalidator"
+)
+
+var defaultNoProxies = []string{".apps-crc.testing", "127.0.0.1", "api.crc.testing", "localhost"}
+
+// ProxyConfig keeps the proxy configuration for the current environment
+type ProxyConfig struct {
+	httpProxy  string
+	httpsProxy string
+	noProxy    []string
+}
+
+// NewProxyConfig creates a proxy configuration with the specified parameters. If a empty string is passed
+// the corresponding environment variable is checked.
+func NewProxyConfig(httpProxy string, httpsProxy string, noProxy string) (*ProxyConfig, error) {
+	defaultScheme := "http"
+
+	// We only parse the Proxy Special char if it is coming from the config file
+	// or user defined flag. In case of Env variable we just ignore parsing.
+	if httpProxy != "" {
+		httpProxy = parseProxySpecialChar(httpProxy, defaultScheme)
+	}
+	if httpsProxy != "" {
+		httpsProxy = parseProxySpecialChar(httpsProxy, defaultScheme)
+	}
+
+	if httpProxy == "" {
+		httpProxy = os.Getenv("http_proxy")
+		if httpProxy == "" {
+			httpProxy = os.Getenv("HTTP_PROXY")
+		}
+	}
+	err := ValidateProxyURL(httpProxy, defaultScheme)
+	if err != nil {
+		return nil, err
+	}
+
+	if httpsProxy == "" {
+		httpsProxy = os.Getenv("https_proxy")
+		if httpsProxy == "" {
+			httpsProxy = os.Getenv("HTTPS_PROXY")
+		}
+	}
+	err = ValidateProxyURL(httpsProxy, defaultScheme)
+	if err != nil {
+		return nil, err
+	}
+
+	np := []string{}
+	np = append(np, defaultNoProxies...)
+
+	if noProxy == "" {
+		noProxy = os.Getenv("no_proxy")
+		if noProxy == "" {
+			noProxy = os.Getenv("NO_PROXY")
+		}
+	}
+
+	if noProxy != "" {
+		np = append(np, strings.Split(noProxy, ",")...)
+	}
+
+	config := ProxyConfig{
+		httpProxy:  httpProxy,
+		httpsProxy: httpsProxy,
+		noProxy:    np,
+	}
+
+	return &config, nil
+}
+
+// ProxyConfig returns a the proxy configuration as a slice, one element for each of the potential settings
+// HTTP_PROXY, HTTPS_PROXY and NO_PROXY. If proxies are not enabled an empty slice is returned.
+func (p *ProxyConfig) ProxyConfig() []string {
+	config := []string{}
```

Comment:
@cfergeau I added it because I thought we will use it when cluster wide proxy part need to implemented but we can add it later if required (removing it now).